ecl/hqlcpp/hqlcset.cpp

```diff
@@ -1424,24 +1415,24 @@ void GeneralSetCursor::buildAssignSelect(BuildCtx & ctx, const CHqlBoundTarget &
     {
         OwnedHqlExpr select = createSelectExpr(LINK(dsIndexExpr), LINK(element));
         if (!cursor->isConditional())
-            translator.buildExprAssign(ctx, target, select);
+            translator.buildExprOrAssign(ctx, target, select, tgt);
         else
         {
             // if (row) tgt = x else tgt = dft;
             BuildCtx subctx(ctx);
```

Comment:
I didn't find a single test case in any regression suite which went through this code!